pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -704,10 +737,39 @@ func (r *ReconcileClusterPool) createCluster(
 			continue
 		}
 		poolRef := poolReference(clp)
+		if clp.Spec.Inventory != nil {
+			poolRef.ClusterDeploymentCustomizationRef = &corev1.LocalObjectReference{Name: cdc.Name}
+		}
 		cd.Spec.ClusterPoolRef = &poolRef
 		lastIndex := len(objs) - 1
 		objs[i], objs[lastIndex] = objs[lastIndex], objs[i]
 	}
+	// Apply inventory customization
+	if clp.Spec.Inventory != nil {
+		for _, obj := range objs {
+			var ok bool
+			if ics, ok = obj.(*corev1.Secret); ok {
+				installConfig, ok := ics.StringData["install-config.yaml"]
+				if ok {
+					newInstallConfig, err := r.getCustomizedInstallConfig([]byte(installConfig), cdc, logger)
+					if err != nil {
+						r.setInventoryValidCondition(clp, false, fmt.Sprintf("failed to customize with %s", cdc.Name), logger)
+						cdc.Status.Conditions = controllerutils.SetClusterDeploymentCustomizationCondition(
+							cdc.Status.Conditions,
+							hivev1.ClusterDeploymentCustomizationAvailableCondition,
+							corev1.ConditionFalse,
+							"CustomizationFailed",
+							"Failed to customize install config",
+							controllerutils.UpdateConditionIfReasonOrMessageChange,
+						)
```

Comment:
I do not see any update on customization for these changes to take effect.

Also, what/who is responsible for making the customization available again when necessary errors are removed?

Reply:
https://github.com/openshift/hive/pull/1687